### PR TITLE
Additions tests for Polygon

### PR DESCRIPTION
**Pull request by z3dev**
Please review the changes.

Not sure why the other commits are listed here...

### CODE REVIEW

**Comment by kaosat-dev** [comment]
More tests is good !
About the extra commits, have you tried rebasing off master ?

File: test/polygon3.js [hunk 1]
```diff
@@ -21,11 +21,83 @@ test('CSG.Polygon3 creates a 3d polygon', t => {
     new Vertex3(new Vector3([0, 10, 0])),
     new Vertex3(new Vector3([0, 10, 10]))
   ]
-  const observed = new Polygon(vertices)
+  var observed = new Polygon(vertices)
```
Comment (kaosat-dev) [inline test/polygon3.js]: Using 'let' here instead of a 'var' is better

File: test/polygon3.js [hunk 2]
```diff
@@ -21,11 +21,83 @@ test('CSG.Polygon3 creates a 3d polygon', t => {
...
+    [0, 10, 0],
+    [0, 10, 10]
+  ]
+  const vectors = [
+    new Vector3([0,  0, 0]),
+    new Vector3([0, 10, 0]),
+    new Vector3([0, 10, 10])
+  ]
+
+  var observed = Polygon.createFromPoints(points)
```
Comment (kaosat-dev) [inline test/polygon3.js]: same here (let > var)

File: test/polygon3.js [hunk 3]
```diff
@@ -21,11 +21,83 @@ test('CSG.Polygon3 creates a 3d polygon', t => {
...
+  const bbox     = observed.boundingBox()
+
+  t.is(volume,0)
+  t.is(area,50)
+  t.deepEqual(features,[0,50])
+  t.deepEqual(bsphere,[new Vector3([0,5,5]),7.0710678118654755])
+  t.deepEqual(bbox,[new Vector3([0,0,0]),new Vector3([0,10,10])])
 })
+
+// checkIfConvex() // throws new Error if not
```
Comment (kaosat-dev) [inline test/polygon3.js]: setup for future tests ?

Comment (z3dev) [inline test/polygon3.js]: notes for more tests

**Pull request merged**